compiler/src/dmd/expressionsem.d

```diff
@@ -5115,7 +5115,13 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 tb = tb.isTypeDArray().next.toBasetype();
             }
 
-            if (global.params.betterC || !sc.needsCodegen())
```

Comment:
Can we stop using betterC as a gate for semantic decision paths and use more granular params?

Reply:
Yeah, we should remove the `betterC` from `dmd.globals.Params` and only use `dmd.globals.Params.use{ModuleInfo,TypeInfo,Exceptions}` in the frontend. The only place that should concern itself with whether `-betterC` was passed on the command-line is the driver and except for seetting `dmd.globals.Params.use{ModuleInfo,TypeInfo,Exceptions}` it is only needed too know whether to link Druntime and Phobos.

Reply:
yes please!!